test/vault/Vault.spec.ts

```diff
@@ -2949,6 +2949,33 @@ describe('Vault', () => {
 
       await vault.connect(admin).exitUnpause();
     });
+
+    it('leaves no dust on withdraw when force is false', async () => {
+      const toDeposit = parseUnits('100');
+      const prevBal = await underlying.balanceOf(alice.address);
+
+      const params = depositParams.build({
+        amount: toDeposit,
+        inputToken: underlying.address,
+        claims: [
+          claimParams.percent(50).to(alice.address).build(),
+          claimParams.percent(50).to(bob.address).build(),
+        ],
+      });
+
+      await vault.connect(alice).deposit(params);
+
+      await moveForwardTwoWeeks();
+      await addYieldToVault('100');
+
+      await vault
+        .connect(alice)
+        .partialWithdraw(alice.address, [2], [parseUnits('1')]);
+
+      const newBal = await underlying.balanceOf(alice.address);
+
+      expect(prevBal.sub(newBal)).to.eq(toDeposit.sub(parseUnits('1')));
+    });
```

Comment:
This test because of the subtraction of 1 done in line 2977 to take the dust into account. Similarly many other tests also had a +1 or -1 in the end to account for the dust. This fix, fixes that issue so now we dont have to account for dust in the tests thus also making the tests more readable and simpler. I have made the changes to the tests and they are passing now.